package/network/services/lldpd/files/lldpd.init

```diff
@@ -19,17 +26,85 @@ find_release_info()
 	echo "${PRETTY_NAME:-Unknown OpenWrt release} @ $(cat /proc/sys/kernel/hostname)"
 }
 
+get_config_restart_hash() {
+	local var="$1"
+	local _string _hash v
+
+	config_load 'lldpd'
+
+	config_get      v 'config' 'lldp_class'; append _string "$v" ","
+	config_get      v 'config' 'agentxsocket'; append _string "$v" ","
+	config_get      v 'config' 'cid_interface'; append _string "$v" ","
+	config_get      v 'config' 'filter'; append _string "$v" ","
+	config_get_bool v 'config' 'readonly_mode'; append _string "$v" ","
+	config_get_bool v 'config' 'lldpmed_no_inventory'; append _string "$v" ","
+	config_get_bool v 'config' 'lldp_no_version'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_lldp' 1; append _string "$v" ","
+	config_get_bool v 'config' 'force_lldp'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_cdp'; append _string "$v" ","
+	config_get      v 'config' 'cdp_version'; append _string "$v" ","
+	config_get_bool v 'config' 'force_cdp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_cdpv2'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_edp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_edp'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_fdp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_fdp'; append _string "$v" ","
+
+	config_get_bool v 'config' 'enable_sonmp'; append _string "$v" ","
+	config_get_bool v 'config' 'force_sonmp'; append _string "$v" ","
+
+	_hash=`echo -n "${_string}" | md5sum | awk '{ print \$1 }'`
+	export -n "$var=$_hash"
+}
+
+get_config_cid_ifaces() {
+	local _ifaces
+	config_get _ifaces 'config' 'cid_interface'
+
+	local _iface _ifnames=""
+	for _iface in $_ifaces; do
+		local _ifname=""
+		if network_get_device _ifname "$_iface" || [ -e "/sys/class/net/$_iface" ]; then
+			append _ifnames "${_ifname:-$_iface}" ","
+		fi
+	done
+
+	export -n "${1}=$_ifnames"
+}
+
 write_lldpd_conf()
 {
-	. /lib/functions/network.sh
-
 	local lldp_description
+	local lldp_hostname
+	local lldp_platform
+	local lldp_tx_interval
+	local lldp_tx_hold
+	local lldp_mgmt_ip
+	local lldp_agetntype
+	local lldp_portidsubtype
+	local lldp_mgmt_ip
+	local lldp_syscapabilities
+	local lldp_location
 
 	config_load 'lldpd'
 	config_get lldp_description 'config' 'lldp_description' "$(find_release_info)"
-
-	local lldp_hostname
 	config_get lldp_hostname 'config' 'lldp_hostname' "$(cat /proc/sys/kernel/hostname)"
+	config_get lldp_platform 'config' 'lldp_platform' ""
+	config_get lldp_tx_interval 'config' 'lldp_tx_interval' 0
+	config_get lldp_tx_hold 'config' 'lldp_tx_hold' 0
+	config_get lldp_mgmt_ip 'config' 'lldp_mgmt_ip' ''
+	config_get lldp_agenttype 'config' 'lldp_agenttype' 'nearest-bridge'
+	config_get lldp_portidsubtype 'config' 'lldp_portidsubtype' 'macaddress'
+	config_get lldp_mgmt_ip 'config' 'lldp_mgmt_ip'
+	config_get lldp_syscapabilities 'config' 'lldp_syscapabilities'
```

Comment:
@systemcrash I noticed that your branch doesn't configure `lldp_syscapabilities` - is that intentional?

Reply:
I don't know where you got that from: my branch leaves `lldp_syscapabilities` untouched. `lldp_syscapabilities` is configured. Line 48-49 + 57.

Reply:
My intention was to remove as much jumble from the noise of the initial PR, to retain focus on what actually (needed to be) changed. You're welcome to include it, but it's just moving lines around which takes a reviewer time to untangle.

Reply:
I must have missed that, I'll recheck.